src/librustc_ast_lowering/expr.rs

```diff
@@ -237,6 +239,7 @@ impl<'hir> LoweringContext<'_, 'hir> {
     }
 
     fn lower_binop(&mut self, b: BinOp) -> hir::BinOp {
+        let span = self.mark_span_with_reason(DesugaringKind::Operator, b.span, None);
```

Comment:
Uh this destroyed Clippy (23 UI tests suddenly failed) and send me on a multiple hour bisecting adventure. We need to enforce test-pass for Clippy ASAP 😄

Reply:
D'oh! Now that we are able to refactor clippy "in situ" that might be more reasonable, not sure. It'd be nice if we could at least get an "advisory" warning perhaps that clippy is affected.

Reply:
@nikomatsakis Normally rustc gates on clippy tests iirc, right now due to some temporary toolstate changes they don't.

Reply:
Given that this PR breaks the ability to understand if something comes from a macro desugaring or not, which is used in both clippy and rustc for diagnostics, I'm inclined to revert this until a solution that can preserve macro desugaring info is found.

Reply:
@Aaron1011 is also working on a fix that resurfaces that, i'm down to landing that as well, but as it currently stands I consider this a rustc bug

Reply:
> Given that this PR breaks the ability to understand if something comes from a macro desugaring or not,

Nit: This was already broken any time we created an `ExpnKind::Desugaring` span. However, `DesugaringKind::Operator` is created much more frequently that other `DesugaringKind`s (or it overlaps with diagnostics/lint-able code in more cases), so it's more of an issue.